src/hooks/UrqlUseMutation.rei

```diff
@@ -1,14 +1,18 @@
 let useMutation:
-  (~request: UrqlTypes.request('response)) =>
-  (
-    UrqlTypes.hookResponse('response),
-    unit => Js.Promise.t(UrqlClient.ClientTypes.operationResult),
-  );
+  UrqlTypes.graphqlDefinition(
+    'parsedResponse,
+    (
+      UrqlTypes.hookResponse('parsedResponse),
+      unit => Js.Promise.t(UrqlClient.ClientTypes.operationResult),
+    ),
+    'functionThatAcceptsVarArgsAndReturnsHookResponse,
```

Comment:
Can we find a more concise name for this type variable? Would `'composeVariables` be appropriate here?

Reply:
it's not really the composeVariables function though. I tried to come up with a name that clearly defines what it is but couldn't come up with anything shorter than this.
I could call also call it `'a`? haha

Reply:
Ah ok, I guess I'm misinterpreting how `composeVariables` works. So the idea is that `composeVariables` returns you a **function**, which accepts your `variables` as a `Js.Json.t`. I guess I'm missing the step around how variables that we pass in as labeled arguments get passed to `composeVariables` as a `Js.Json.t`:

```reason
Hooks.useMutation(MyMutation.definition, /* How are we able to pass these in and get a Js.Json.t? */ ~var1=x, ~var2=y, ());
```

Is the PPX doing this magic for us? It looks like it from your description in #132. If that's the case, yeah I agree it's difficult to find a real name for this. I'll keep thinking this one over, just wanted to make sure I understand what this is doing.

Reply:
composeVariables (the thing provided by the ppx), is a higher-order function that accepts a function from Js.Json.t => 'a and returns a new function that accepts all the graphql vars as named args and then calls your passed-in function with a Js.Json.t and returns the result.

We use this by wrapping our hook internals in a function that accepts variables, and then passing this to composeVarialbes and returning the result. This makes useMutation a function that accepts one argument and returns a function of `(~var1, ~var2... ()) => hookResponse`. Because of the magic of auto-currying etc, the user doesn't see this and it feels like they're just calling one function called useMutation.

---

Another way to explain it is to say that the simplest composeVariables function would be something like this:
```
let composeVariables = (f, v) => f(v);
Hooks.useMutation((a => a, "query", composeVariables), Js.Json.null); 
```

useMutation partially applies the composeVariables we pass in with the function that should be called with the variables object, then returns the a function that still needs to be passed `v`. We then immediately do so (`Js.Json.null`) in our call to "useMutation".
Passing `Js.Json.null` fully applies the `composeVariables` we passed in, causing useMutation's callback to get called, which returns the hookResponse.

Reply:
Thanks for the detailed explanation – I always forget about auto-currying in Reason spending so much of my day in JS land. That all makes perfect sense 🎉 A very clever implementation and a great usage of the benefits of an FP language!